OPHD/Population/Population.cpp

```diff
@@ -98,156 +99,146 @@ int Population::adults() const
 }
 
 
 void Population::spawn_retiree()
 {
 	int total_adults = mPopulation[PersonRole::ROLE_WORKER] + mPopulation[PersonRole::ROLE_SCIENTIST];
-	if (total_adults > 0)
-	{
-		mPopulationGrowth[PersonRole::ROLE_RETIRED] += total_adults / 10;
 
-		int divisor = std::max(total_adults, adultToRetireeBase);
+	if (total_adults < 1) { return; }
 
-		divisor = ((divisor / 40) * 3 + 40) * 4;
+	mPopulationGrowth[PersonRole::ROLE_RETIRED] += total_adults / 10;
 
-		int retiree = mPopulationGrowth[PersonRole::ROLE_RETIRED] / divisor;
-		mPopulationGrowth[PersonRole::ROLE_RETIRED] = mPopulationGrowth[PersonRole::ROLE_RETIRED] % divisor;
+	int divisor = std::max(total_adults, adultToRetireeBase);
 
-		mPopulation[PersonRole::ROLE_RETIRED] += retiree;
+	divisor = ((divisor / 40) * 3 + 40) * 4;
 
-		/** Workers retire earlier than scientists. */
-		if (randomNumber.generate(0, 100) <= 45) { if (mPopulation[PersonRole::ROLE_SCIENTIST] > 0) { mPopulation[PersonRole::ROLE_SCIENTIST] -= retiree; } }
-		else { if (mPopulation[PersonRole::ROLE_WORKER] > 0) { mPopulation[PersonRole::ROLE_WORKER] -= retiree; } }
-	}
+	int retiree = mPopulationGrowth[PersonRole::ROLE_RETIRED] / divisor;
+	mPopulationGrowth[PersonRole::ROLE_RETIRED] = mPopulationGrowth[PersonRole::ROLE_RETIRED] % divisor;
+
+	mPopulation[PersonRole::ROLE_RETIRED] += retiree;
+
+	/** Workers retire earlier than scientists. */
+	if (randomNumber.generate(0, 100) <= 45) { if (mPopulation[PersonRole::ROLE_SCIENTIST] > 0) { mPopulation[PersonRole::ROLE_SCIENTIST] -= retiree; } }
+	else { if (mPopulation[PersonRole::ROLE_WORKER] > 0) { mPopulation[PersonRole::ROLE_WORKER] -= retiree; } }
 }
 
 
 void Population::kill_children(int morale, int nurseries)
 {
-	if (mPopulation[PersonRole::ROLE_CHILD] > 0)
-	{
-		mPopulationDeath[PersonRole::ROLE_CHILD] += mPopulation[PersonRole::ROLE_CHILD];
+	if (mPopulation[PersonRole::ROLE_CHILD] < 1) { return; }
 
-		int divisor = moraleModifierTable[moraleIndex(morale)].mortalityRate + (nurseries * 10);
+	mPopulationDeath[PersonRole::ROLE_CHILD] += mPopulation[PersonRole::ROLE_CHILD];
 
-		int deaths = mPopulationDeath[PersonRole::ROLE_CHILD] / divisor;
-		mPopulationDeath[PersonRole::ROLE_CHILD] = mPopulationDeath[PersonRole::ROLE_CHILD] % divisor;
+	int divisor = moraleModifierTable[moraleIndex(morale)].mortalityRate + (nurseries * 10);
 
-		mPopulation[PersonRole::ROLE_CHILD] -= deaths;
-		mDeathCount += deaths;
+	int deaths = mPopulationDeath[PersonRole::ROLE_CHILD] / divisor;
+	mPopulationDeath[PersonRole::ROLE_CHILD] = mPopulationDeath[PersonRole::ROLE_CHILD] % divisor;
 
-		if (mPopulation[PersonRole::ROLE_CHILD] <= 0)
-		{
-			mPopulationDeath[PersonRole::ROLE_CHILD] = 0;
-			mPopulationGrowth[PersonRole::ROLE_STUDENT] = 0;
-		}
+	mPopulation[PersonRole::ROLE_CHILD] -= deaths;
+	mDeathCount += deaths;
+
+	if (mPopulation[PersonRole::ROLE_CHILD] <= 0)
+	{
+		mPopulationDeath[PersonRole::ROLE_CHILD] = 0;
+		mPopulationGrowth[PersonRole::ROLE_STUDENT] = 0;
 	}
```

Comment:
This block will only run for deaths calculated here. If deaths are the result of some external cause, this block won't run due to the earlier check at the start of the function. That may or may not be a problem. In particular, if the crime module allows for murder, you might get odd effects. In particular, a new child could be near instantly transformed to a student.

This concern applies to multiple blocks from various population functions.

Reply:
Flagged this PR with 996 to track as possible issue.